src/nuc/selector.py

```diff
@@ -64,16 +96,34 @@ def apply(self, value: Dict[str, Any]) -> Any:
         value
             The dict that this selector should be applied to.
         """
+        match self.target:
+            case SelectorTarget.TOKEN:
+                return self._apply_on_value(self.path, value)
+            case SelectorTarget.CONTEXT:
+                if not self.path:
+                    return None
+                first = self.path[0]
+                context_value = context.get(first)
```

Comment:
I think we should also check here if user Selector is correct for a given value type; if value is scalar instead of object this silently returns `None` right? better to return error i think..

Reply:
Is this the same comment as in the nilvm repo? Can you provide an example with a policy that misbehaves? In general the lookup will return something and the policy gets applied over that so if it returns the wrong thing it'll be fine,

Reply:
yes, we already cleared in previous related PR. thanks